main.cc

```diff
@@ -1496,12 +1496,18 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
             }
             ss.local().register_protocol_server(redis_ctl);
 
-            //FIXME: assumes that the CQL server exists, which is not necessarily the case
-            websocket::controller websocket_ctl(cql_server_ctl.server());
-            with_scheduling_group(dbcfg.statement_scheduling_group, [&websocket_ctl] {
-                return websocket_ctl.start_server();
-            }).get();
-            ss.local().register_protocol_server(websocket_ctl);
+            websocket::controller websocket_ctl(cql_server_ctl.server(), *cfg);
```

Comment:
Please, don't feed `db::config` directory into controller. Better add a `websocket::server::config`, initialize it with needed data from `db::config` and move the value into controller.

Reply:
@xemul side note, I just realized that the author most likely did it because we have a precedent - database::config is fed directly to the cql controller already. So perhaps we might want to rethink that one too?

Reply:
@psarna , for transpot it was done for a reason -- the server can be restarted via API and by that time config may change, so it should pick up the new values ... somehow